src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -406,19 +408,36 @@ private void updateTableSchema(TableReference tableRef) {
       FieldList sourceFields = sourceTable.getDefinition().getSchema().getFields();
       tableFieldsList = sourceFields.stream().map(Field::getName).collect(Collectors.toList());
       FieldList destinationFields = destinationTable.getDefinition().getSchema().getFields();
+      Map<String, Field> destinationFieldMap = destinationFields.stream()
+        .collect(Collectors.toMap(Field::getName, x -> x));
 
-      List<String> fieldsName = destinationFields.stream().map(Field::getName).collect(Collectors.toList());
-
-      List<Field> newFields = sourceFields.stream().filter(field -> !fieldsName.contains(field.getName()))
+      List<Field> newFields = sourceFields.stream().filter(field -> destinationFieldMap.get(field.getName()) == null)
         .collect(Collectors.toList());
+      List<Field> updateFields = new ArrayList<>();
+      sourceFields.stream()
+        .filter(sourceField -> sourceField.getMode() == null
+          || (sourceField.getMode() != null && sourceField.getMode().equals(Field.Mode.NULLABLE)))
+        .filter(sourceField -> destinationFieldMap.get(sourceField.getName()) != null)
```

Comment:
it would be better for this to be a map to get the destinationField, followed by a filter on null and the field mode, followed by a collect() instead of a forEach.

Reply:
Done.